src/averageAges.js

```diff
@@ -55,10 +73,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const childrens = people.filter(person => person.mother !== null
+    && (onlyWithSon
+      ? person.sex === SEX_MALE
+      : true
+    ));
+
+  const ageDifferences = childrens.map(({ mother: motherName, born }) => {
+    const mother = people.find(({ name }) => name === motherName);
+
+    return mother ? born - mother.born : null;
+  }).filter(element => element !== null);
```

Comment:
.filter(element => element) 

you don't need !== null here, it should work the same way